VMRecovery/Testing/Create-TestVM.ps1

```diff
@@ -0,0 +1,848 @@
+        [string]$rescueDiskUri,
+        [switch]$useManagedDisk,
+        [string]$accountType = 'StandardLRS'
+    )
+
+    $storageAccountKey = ($storageAccount | Get-AzureRmStorageAccountKey )[0].Value
+    $storageContext = New-AzureStorageContext -StorageAccountName $storageAccount.storageAccountName -StorageAccountKey $storageAccountKey
+    $containerName = $rescueDiskUri.Split('/')[-2]
+    if(!(Get-AzureStorageContainer -Context $storageContext -Name $containerName -ErrorAction SilentlyContinue))
+    {
+        $container = New-AzureStorageContainer -Context $storageContext -Name $containerName
+    }
+
+    show-progress "Copying rescue disk VHD into storage account $($storageAccount.storageAccountName)"
+    $rescueDiskBlobName = $rescueDiskUri.Split('/')[-1]
+    $rescueDiskCopyDiskName = "$($rescueDiskBlobName.Split('.')[0])$vmName"
+    $rescueDiskCopyBlobName = "$rescueDiskCopyDiskName.vhd"
+    $blobCopyStartTime = get-date
+    $rescueDiskBlobCopy = Start-AzureStorageBlobCopy -AbsoluteUri $rescueDiskUri -DestContainer $containerName -DestBlob $rescueDiskCopyBlobName -DestContext $storageContext -force
+    $rescueDiskBlobCopyUri = $rescueDiskBlobCopy.ICloudBlob.Uri
+    $rescueDiskBlobCopyStatus = (Get-AzureStorageBlobCopyState -CloudBlob $rescueDiskBlobCopy.ICloudBlob -Context $storageContext -WaitForComplete -ServerTimeoutPerRequest 60).Status
+
+    <#
+    $timeout = 60
+    do {
+        $secondsInterval = 5
+        start-sleep -Seconds $secondsInterval
+        $secondsElapsed += $secondsInterval
+        $rescueDiskBlobCopyStatus = (Get-AzureStorageBlobCopyState -CloudBlob $rescueDiskBlobCopy.ICloudBlob -Context $storageContext).Status
+    } until (($rescueDiskBlobCopyStatus -eq 'Success') -or ($secondsElapsed -ge $timeout))
+    #>
+
+    if ($rescueDiskBlobCopyStatus -eq 'Success')
+    {
+        $blobCopyDuration = [Math]::Round((new-timespan -start $blobCopyStartTime -end (get-date)).TotalSeconds,2)
+        show-progress "Rescue disk copied to $rescueDiskBlobCopyUri"
+        show-progress "Rescue disk copy completed in $blobCopyDuration seconds"
+    }
+    else
+    {
+        show-progress "Copied failed: $rescueDiskBlobCopyStatus"
+        exit
+    }
+
+    if($useManagedDisk)
+    {
+        show-progress "Creating managed disk from copied rescue disk VHD"
+        $managedDiskConfig = New-AzureRmDiskConfig -AccountType $accountType -Location $location -CreateOption Import -StorageAccountId $storageAccount.Id -SourceUri $rescueDiskBlobCopyUri
+        $managedDisk = New-AzureRmDisk -Disk $managedDiskConfig -ResourceGroupName $resourceGroupName -DiskName $rescueDiskCopyDiskName
+        show-progress "Created managed disk $($managedDisk.Name)"
+        return $managedDisk
+    }
+    else
+    {
+        return $rescueDiskBlobCopyUri.AbsoluteUri
+    }
+}
+function Get-VMName
+{
+    param(
+        [string]$skus,
+        [string]$offer
+    )
+
+    $vmNamePrefix = ''
+
+    switch -regex ($skus)
```

Comment:
Doesn't look like you are actually doing a RegEx.  Perhaps this mapping is better modeled as a hashtable and then you just index into it?  Then you could use the same for the offer logic?  If you decided to keep the switch you might consider that many of them are just a lowercase of the value and that if you add a break statement you can improve execution time a bit more.